### PR TITLE
update jobs guide with examples

### PR DESCRIPTION
**Pull request by rosieyohannan**
ticket: https://circleci.atlassian.net/browse/DOCSTEAM-176?atlOrigin=eyJpIjoiNWZlZjdjMDJjN2JhNDQwY2JjNTE2NTk2MjVmYzRiYTQiLCJwIjoiaiJ9

reinstate jobs and steps overview in sidebar and update usage examples.

I'm doing this now because I need somewhere to link to for another ticket I'm working on: https://circleci.atlassian.net/browse/DOCSTEAM-32?atlOrigin=eyJpIjoiMTQwN2M0ODRlNjRlNDQyYWFiMzk1ZjRhNjRjZjBkNzYiLCJwIjoiaiJ9

### CODE REVIEW

File: jekyll/_cci2/jobs-steps.md [hunk 1]
```diff
@@ -1,30 +1,22 @@
 ---
 layout: classic-docs
-title: "Orbs, Jobs, Steps, and Workflows"
-short-title: "Orbs, Jobs, Steps, and Workflows"
-description: "Description of Jobs and Steps"
-categories: [migration]
-order: 2
+title: "Jobs and Step"
```
Comment (rosieyohannan) [inline jekyll/_cci2/jobs-steps.md:3]:
```suggestion
title: "Jobs and Steps"
```

File: jekyll/_cci2/jobs-steps.md [hunk 2]
```diff
@@ -33,89 +25,110 @@ The following diagram illustrates how data flows between jobs:
 
 ![Jobs Overview]( {{ site.baseurl }}/assets/img/docs/jobs-overview.png)
 
-Jobs can be run using the `machine` (linux), macOS or Windows executors, or the `docker` executor, which can compose Docker containers to run your jobs and any services they require, such as databases.
+**Note**: The jobs names shown in this diagram are just examples, you can call your jobs whatever you want.
```
Comment (devinmarieb) [inline jekyll/_cci2/jobs-steps.md:28]: I know this is technically correct, but "jobs names" is just so awkward to read/say. Maybe "the names of the jobs"?

File: jekyll/_cci2/jobs-steps.md [hunk 3]
```diff
@@ -33,89 +25,110 @@ The following diagram illustrates how data flows between jobs:
...
+
+Jobs can be run in docker containers, using the Docker executor, or in virtual machines using the `machine` executor, with linux, macOS or Windows images. Secondary containers or VMs can be configures to to attached services, such as databases, to run alongside your jobs.
 
-When using the `docker` executor the container images listed under the `docker:` keys specify the containers to start. Any public Docker images can be used with the `docker` executor.
+When using the Docker executor, images listed under the `docker:` key specifies the containers you want to start for your job. Any public Docker images can be used with the Docker executor but CircleCI provides convenience images for a variety of use-cases. For full lists of available convevience and VM images, see the [CircleCI Developer Hub](https://circleci.com/developer/images).
```
Comment (devinmarieb) [inline jekyll/_cci2/jobs-steps.md:32]:
```suggestion
When using the Docker executor, images listed under the `docker:` key specify the containers you want to start for your job. Any public Docker images can be used with the Docker executor, but CircleCI provides convenience images for a variety of use-cases. For full lists of available convenience and VM images, see the [CircleCI Developer Hub](https://circleci.com/developer/images).
```

File: jekyll/_cci2/jobs-steps.md [hunk 4]
```diff
@@ -1,30 +1,22 @@
...
-## Orbs overview
-{: #orbs-overview }
-
-Orbs are packages of config that you either import by name or configure inline to simplify your config, share, and reuse config within and across projects. See [Orbs Concepts]({{ site.baseurl }}/2.0/orb-concepts/) for details about how to use orbs in your config and an introduction to orb design. Visit the [Orbs Registry](https://circleci.com/developer/orbs) to search for orbs to help simplify your config.
-
 ## Jobs overview
 {: #jobs-overview }
 
-Jobs are collections of steps. All of the steps in the job are executed in a single unit, either within a fresh container or VM.
+A CircleCI job is a collection of steps. All of the steps in the job are executed in a single unit, either within a fresh container or a virtual machine. Jobs are orchestrated using [workflows]({{ site.baseurl }}/2.0/workflows/).
```
Comment (devinmarieb) [inline jekyll/_cci2/jobs-steps.md:19]:
```suggestion
A CircleCI job is a collection of steps. All of the steps in the job are executed in a single unit, either within a fresh container, or a virtual machine. Jobs are orchestrated using [workflows]({{ site.baseurl }}/2.0/workflows/).
```

File: jekyll/_cci2/jobs-steps.md [hunk 5]
```diff
@@ -33,89 +25,110 @@ The following diagram illustrates how data flows between jobs:
...
+  my-workflow:
+    jobs:
+      - print-a-message:
+          message: Hello!
+```
+
+## Using a job from an orb
+{: #using-a-job-from-an-orb }
+
+Orbs are packages or reusable config that you can use in your project configuration. Orbs usually contain both commands that you can use in your jobs, and whole jobs that you can schedule in your workflows. Take the [Slack orb](https://circleci.com/developer/orbs/orb/circleci/slack) as an example. This orb provides a job called [`on-hold`](https://circleci.com/developer/orbs/orb/circleci/slack#usage-on_hold_notification) which you can ue in your workflows. This job pauses the workflow to require manual approval and send a slack notification. To use this job just reference it in your workflow (see line 10):
```
Comment (devinmarieb) [inline jekyll/_cci2/jobs-steps.md:69]: I think the word "both" here is confusing. Maybe it's just me but I read it as an orb should contain "two" commands, as well as whole jobs.

Comment (devinmarieb) [inline jekyll/_cci2/jobs-steps.md:69]:
```suggestion
Orbs are packages or reusable config that you can use in your project configuration. Orbs usually contain both commands that you can use in your jobs, and whole jobs that you can schedule in your workflows. Take the [Slack orb](https://circleci.com/developer/orbs/orb/circleci/slack) as an example. This orb provides a job called [`on-hold`](https://circleci.com/developer/orbs/orb/circleci/slack#usage-on_hold_notification) which you can use in your workflows. This job pauses the workflow to require manual approval and send a slack notification. To use this job just reference it in your workflow (see line 10):
```

**Review by devinmarieb — CHANGES_REQUESTED** [review]
Some minor comments and a few suggestions for spelling, etc!

File: jekyll/_cci2/jobs-steps.md (hunk 2 above)
Comment (rosieyohannan) [reply jekyll/_cci2/jobs-steps.md:28]: I actually think that was a mistake on my part! Thanks for spotting

Comment (rosieyohannan) [inline jekyll/_cci2/jobs-steps.md:28]:
```suggestion
**Note**: The job names shown in this diagram are just examples, you can call your jobs whatever you want.
```

File: jekyll/_cci2/jobs-steps.md [hunk 6]
```diff
@@ -33,89 +25,110 @@ The following diagram illustrates how data flows between jobs:
...
+            - slack/on-hold
+          type: approval
+      - my_deploy_job:
+          requires:
+            - pause_workflow
 ```
 
-## Sample configuration with concurrent jobs
-{: #sample-configuration-with-concurrent-jobs }
+## Using a command from an orb in a job
```
Comment (rosieyohannan) [inline jekyll/_cci2/jobs-steps.md:95]:
```suggestion
## Using a command from an orb in a job
{: #using-a-command-from-an-orb-in-a-job }
```

File: jekyll/_cci2/jobs-steps.md [hunk 7]
```diff
@@ -33,89 +25,111 @@ The following diagram illustrates how data flows between jobs:
...
+  my-workflow:
+    jobs:
+      - print-a-message:
+          message: Hello!
+```
+
+## Using a job from an orb
+{: #using-a-job-from-an-orb }
+
+Orbs are packages or reusable config that you can use in your project configuration. Orbs usually contain both commands that you can use in your jobs, and whole jobs that you can schedule in your workflows. Take the [Slack orb](https://circleci.com/developer/orbs/orb/circleci/slack) as an example. This orb provides a job called [`on-hold`](https://circleci.com/developer/orbs/orb/circleci/slack#usage-on_hold_notification) which you can use in your workflows. This job pauses the workflow to require manual approval and send a slack notification. To use this job just reference it in your workflow (see line 10):
```
Comment (rosieyohannan) [inline jekyll/_cci2/jobs-steps.md:69]:
```suggestion
Orbs are packages or reusable config that you can use in your project configuration. Orbs usually contain commands that you can use in your jobs, and whole jobs that you can schedule in your workflows. Take the [Slack orb](https://circleci.com/developer/orbs/orb/circleci/slack) as an example. This orb provides a job called [`on-hold`](https://circleci.com/developer/orbs/orb/circleci/slack#usage-on_hold_notification) which you can use in your workflows. This job pauses the workflow to require manual approval and send a slack notification. To use this job just reference it in your workflow (see line 10):
```

File: jekyll/_cci2/jobs-steps.md [hunk 8]
```diff
@@ -33,89 +25,111 @@ The following diagram illustrates how data flows between jobs:
 
 ![Jobs Overview]( {{ site.baseurl }}/assets/img/docs/jobs-overview.png)
 
-Jobs can be run using the `machine` (linux), macOS or Windows executors, or the `docker` executor, which can compose Docker containers to run your jobs and any services they require, such as databases.
+**Note**: The job names shown in this diagram are just examples, you can call your jobs whatever you want.
```
Comment (rosieyohannan) [inline jekyll/_cci2/jobs-steps.md:28]:
```suggestion
**Note**: The job names shown in this diagram are just examples, you can name your jobs whatever you want.
```

File: jekyll/_cci2/jobs-steps.md [hunk 9]
```diff
@@ -33,89 +25,111 @@ The following diagram illustrates how data flows between jobs:
 
 ![Jobs Overview]( {{ site.baseurl }}/assets/img/docs/jobs-overview.png)
 
-Jobs can be run using the `machine` (linux), macOS or Windows executors, or the `docker` executor, which can compose Docker containers to run your jobs and any services they require, such as databases.
+**Note**: The job names shown in this diagram are just examples, you can name your jobs whatever you want.
+
+Jobs can be run in docker containers, using the Docker executor, or in virtual machines using the `machine` executor, with linux, macOS or Windows images. Secondary containers or VMs can be configures to to attached services, such as databases, to run alongside your jobs.
```
Comment (rosieyohannan) [inline jekyll/_cci2/jobs-steps.md:30]:
```suggestion
Jobs can be run in docker containers, using the Docker executor, or in virtual machines using the `machine` executor, with linux, macOS or Windows images. Secondary containers or VMs can be configured to attach services, such as databases, to run alongside your jobs.
```

File: jekyll/_cci2/jobs-steps.md [hunk 10]
```diff
@@ -33,89 +25,111 @@ The following diagram illustrates how data flows between jobs:
...
 
-Jobs can be run using the `machine` (linux), macOS or Windows executors, or the `docker` executor, which can compose Docker containers to run your jobs and any services they require, such as databases.
+**Note**: The job names shown in this diagram are just examples, you can name your jobs whatever you want.
+
+Jobs can be run in docker containers, using the Docker executor, or in virtual machines using the `machine` executor, with linux, macOS or Windows images. Secondary containers or VMs can be configured to attach services, such as databases, to run alongside your jobs.
 
-When using the `docker` executor the container images listed under the `docker:` keys specify the containers to start. Any public Docker images can be used with the `docker` executor.
+When using the Docker executor, images listed under the `docker:` key specify the containers you want to start for your job. Any public Docker images can be used with the Docker executor, but CircleCI provides convenience images for a variety of use-cases. For full lists of available convenience and VM images, see the [CircleCI Developer Hub](https://circleci.com/developer/images).
```
Comment (rosieyohannan) [inline jekyll/_cci2/jobs-steps.md:32]:
```suggestion
When using the Docker executor, images listed under the `docker:` key specify the containers you want to start for your job. Any public Docker images can be used with the Docker executor, but CircleCI provides convenience images for a variety of use-cases. Full lists of available convenience and VM images are available in the [CircleCI Developer Hub](https://circleci.com/developer/images).
```

File: jekyll/_cci2/jobs-steps.md [hunk 11]
```diff
@@ -33,89 +25,111 @@ The following diagram illustrates how data flows between jobs:
...
 
 See the [Choosing an Executor Type]({{ site.baseurl }}/2.0/executor-types/) document for use cases and comparisons of the different executor types.
 
 ## Steps overview
 {: #steps-overview }
 
-Steps are a collection of executable commands which are run during a job, the `checkout:` key is required to checkout your code and a key for `run:` enables addition of arbitrary, multi-line shell command scripting.  In addition to the `run:` key, keys for `save_cache:`, `restore_cache:`,  `deploy:`, `store_artifacts:`, `store_test_results:` and `add_ssh_keys` are nested under Steps.
+Steps are a collection of executable commands which are run during a job. The `checkout:` key is required to checkout your code and the `run:` key enables addition of arbitrary, multi-line shell command scripting.  In addition to the `run:` key, keys for `save_cache:`, `restore_cache:`, `store_artifacts:`, `store_test_results:`, and `add_ssh_keys` are nested under Steps. For a full list of step options see the [Configuration Reference Steps Key]({{ site.baseurl }}/2.0/configuration-reference/#steps).
```
Comment (rosieyohannan) [inline jekyll/_cci2/jobs-steps.md:39]:
```suggestion
Steps are collections of executable commands, which are run during a job. The `checkout:` key is required under `steps` to checkout your code and the `run:` key enables addition of arbitrary, multi-line shell command scripting.  In addition to the `run:` key, keys for `save_cache:`, `restore_cache:`, `store_artifacts:`, `store_test_results:`, and `add_ssh_keys` are nested under Steps. For a full list of step options see the [Configuration Reference Steps Key]({{ site.baseurl }}/2.0/configuration-reference/#steps).
```

File: jekyll/_cci2/jobs-steps.md [hunk 12]
```diff
@@ -33,89 +25,111 @@ The following diagram illustrates how data flows between jobs:
...
+Steps are collections of executable commands, which are run during a job. The `checkout:` key is required under `steps` to checkout your code and the `run:` key enables addition of arbitrary, multi-line shell command scripting.  In addition to the `run:` key, keys for `save_cache:`, `restore_cache:`, `store_artifacts:`, `store_test_results:`, and `add_ssh_keys` are nested under Steps. For a full list of step options see the [Configuration Reference Steps Key]({{ site.baseurl }}/2.0/configuration-reference/#steps).
 
-## Sample configuration with imported orb
-{: #sample-configuration-with-imported-orb }
+## Passing parameters to jobs
+{: #passing-parameters-to-jobs }
 
-Find full details of the AWS S3 orb in the [CircleCI Orbs Registry](https://circleci.com/developer/orbs/orb/circleci/aws-s3#commands-sync).
+Using parameters you can run a single job multiple times with different outcomes. An extension of this functionality is [matrix jobs]({{site.baseurl}}/2.0/configuration-reference/#matrix-requires-version-21). Below is a basic example of passing a parameter to a job when it is run.
```
Comment (rosieyohannan) [inline jekyll/_cci2/jobs-steps.md:44]:
```suggestion
Using parameters allows you to run a single job multiple times for different scenarios, such as different package versions or execution environments. An extension of this functionality is [matrix jobs]({{site.baseurl}}/2.0/configuration-reference/#matrix-requires-version-21). Below is a basic example of passing a parameter to a job when it is run.
```

File: jekyll/_cci2/jobs-steps.md [hunk 13]
```diff
@@ -33,89 +25,111 @@ The following diagram illustrates how data flows between jobs:
...
+  my-workflow:
+    jobs:
+      - print-a-message:
+          message: Hello!
+```
+
+## Using a job from an orb
+{: #using-a-job-from-an-orb }
+
+Orbs are packages or reusable config that you can use in your project configuration. Orbs usually contain commands that you can use in your jobs, and whole jobs that you can schedule in your workflows. Take the [Slack orb](https://circleci.com/developer/orbs/orb/circleci/slack) as an example. This orb provides a job called [`on-hold`](https://circleci.com/developer/orbs/orb/circleci/slack#usage-on_hold_notification) which you can use in your workflows. This job pauses the workflow to require manual approval and send a slack notification. To use this job just reference it in your workflow (see line 10):
```
Comment (rosieyohannan) [inline jekyll/_cci2/jobs-steps.md:69]:
```suggestion
Orbs are packages or reusable configuration that you can use in your projects. Orbs usually contain commands that you can use in your jobs, and whole jobs that you can schedule in your workflows. Take the [Slack orb](https://circleci.com/developer/orbs/orb/circleci/slack) as an example. This orb provides a job called [`on-hold`](https://circleci.com/developer/orbs/orb/circleci/slack#usage-on_hold_notification), which you can use in your workflows. This job pauses the workflow to require manual approval, and sends a slack notification. To use this job, just reference it in your workflow (see line 10):
```

File: jekyll/_cci2/jobs-steps.md [hunk 14]
```diff
@@ -33,89 +25,111 @@ The following diagram illustrates how data flows between jobs:
...
+          requires:
+            - pause_workflow
 ```
 
-## Sample configuration with concurrent jobs
-{: #sample-configuration-with-concurrent-jobs }
+## Using a command from an orb in a job
+{: #using-a-command-from-an-orb-in-a-job }
+
+Using the [Slack orb](https://circleci.com/developer/orbs/orb/circleci/slack) as an example again, the orb includes a command called `notify`, to notify a specified slack channel. You can reference this command in your job as follows (see line 16):
```
Comment (rosieyohannan) [inline jekyll/_cci2/jobs-steps.md:98]:
```suggestion
Using the [Slack orb](https://circleci.com/developer/orbs/orb/circleci/slack) as an example again, this orb includes a command called `notify`, which is used to notify a specified slack channel. You can reference this command in your job as follows (see line 16):
```

File: jekyll/_cci2/jobs-steps.md [hunk 15]
```diff
@@ -33,89 +25,111 @@ The following diagram illustrates how data flows between jobs:
...
 
-Jobs can be run using the `machine` (linux), macOS or Windows executors, or the `docker` executor, which can compose Docker containers to run your jobs and any services they require, such as databases.
+**Note**: The job names shown in this diagram are just examples, you can name your jobs whatever you want.
+
+Jobs can be run in docker containers, using the Docker executor, or in virtual machines using the `machine` executor, with linux, macOS or Windows images. Secondary containers or VMs can be configured to attach services, such as databases, to run alongside your jobs.
 
-When using the `docker` executor the container images listed under the `docker:` keys specify the containers to start. Any public Docker images can be used with the `docker` executor.
+When using the Docker executor, images listed under the `docker:` key specify the containers you want to start for your job. Any public Docker images can be used with the Docker executor, but CircleCI provides convenience images for a variety of use-cases. Full lists of available convenience and VM images are available in the [CircleCI Developer Hub](https://circleci.com/developer/images).
```
Comment (rosieyohannan) [inline jekyll/_cci2/jobs-steps.md:32]:
```suggestion
When using the Docker executor, images listed under the `docker` key specify the containers you want to start for your job. Any public Docker images can be used with the Docker executor, but CircleCI provides convenience images for a variety of use-cases. Full lists of available convenience and VM images are available in the [CircleCI Developer Hub](https://circleci.com/developer/images).
```

**Review by devinmarieb — APPROVED** [review]

**Pull request merged**